src/briefcase/commands/new.py

```diff
@@ -103,10 +91,11 @@ def validate_app_name(self, candidate):
         :returns: True. If there are any validation problems, raises ValueError
             with a diagnostic message.
         """
-        if not PEP508_NAME_RE.match(candidate):
+        if not re.match('^[a-z][a-zA-Z0-9._-]*[a-zA-Z0-9]$', candidate):
```

Comment:
If this is the correct regex pattern for PEP508 compliance, we should update the PEP508 regex, not define a complex new regex on the fly.

Reply:
I went ahead and updated the pep508 regex.